src/functions/InModuleScope.ps1

```diff
@@ -78,45 +78,47 @@
     $sessionState = Set-SessionStateHint -PassThru -Hint "Module - $($module.Name)" -SessionState $module.SessionState
 
     $wrapper = {
-        param (
-            [Parameter(Mandatory = $true)]
-            [scriptblock]
-            ${Script Block},
-
-            [hashtable]
-            $___Parameters___ = @{ },
-
-            [object[]]
-            $___ArgumentList___ = @(),
-
-            [System.Management.Automation.SessionState]
-            ${Session State},
-
-            ${Set Dynamic Parameter Variable}
-        )
+        param ($private:______inmodule_parameters)
 
         # This script block is used to create variables for provided parameters that
         # the real scriptblock can inherit. Makes defining a param-block optional.
 
-        & ${Set Dynamic Parameter Variable} -SessionState ${Session State} -Parameters $___Parameters___
+        foreach ($private:______current in $private:______inmodule_parameters.Parameters.GetEnumerator()) {
+            $private:______inmodule_parameters.SessionState.PSVariable.Set($private:______current.Key, $private:______current.Value)
+        }
+
+        # Splatting expressions isn't allowed. Assigning to new private variable
+        $private:______arguments = $private:______inmodule_parameters.ArgumentList
+        $private:______parameters = $private:______inmodule_parameters.Parameters
+
+        if ($private:______parameters.Count -gt 0) {
+            & $private:______inmodule_parameters.ScriptBlock @private:______parameters @private:______arguments
```

Comment:
Change the order here please.

Reply:
of the splats or flip the if/else?

Reply:
The splatting. In WindowsPowerShell you will get the parameters as first in the args array if you splat them first and there is no param block: 

```powershell

$p = @{ a = 10 }
$a = 1, 2

& {
    $args[0]
} @p @a

& {
    $args[0]
} @a @p

# output 
-a:
1
```

Reply:
Updated, but it feels wrong considering it breaks with previous behavioro and how PowerShell does it.

Reply:
PowerShell does it correctly in v7, but not in v5. If you specified params you would not see change in behavior with this approach. It is more that the previous version of InModuleScope does it incorrectly and you are now fixing it. :) If someone was depending on $args[0] they would not be able to get them consistently across versions and we would have an issue for that already I think.

Reply:
Aha. That must be 7.1 fix. I see it working as you say in 7.1.3, while 7.0.6 (used in devcontainer atm.) behaves the same as 5.1. 👍

Reply:
kk did not know that.